tools/python/create_nifi_reporting_task.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+"""This is a script to create a ReportingTask in NiFi via the REST API"""
+import sys
+import argparse
+import nipyapi
+
+# no stack trace
+sys.tracebacklimit = 0
+
+
+def init(url: str, username: str, password: str, ca_file: str):
+    """Initialize authenticated connection to NiFi"""
+    nipyapi.config.nifi_config.host = url
+    nipyapi.security.set_service_ssl_context(service='nifi', ca_file=ca_file)
+
+    try:
+        nipyapi.security.service_login(service='nifi', username=username, password=password)
+        print("Successfully authenticated and established connection with [%s]" % url)
+    except Exception as ex:
+        raise Exception("Failed to connect to {}: {}".format(url, str(ex))) from None
+
+
+def find_reporting_task(name: str, port: str):
+    """Find a ReportingTask via its name and port"""
+    flow_api = nipyapi.nifi.apis.flow_api.FlowApi()
+
+    try:
+        reporting_tasks = flow_api.get_reporting_tasks().reporting_tasks
+    except Exception as ex:
+        raise Exception("Could not find ReportingTask[{}/{}]: {}"
+                        .format(name, port, str(ex))) from None
+
+    for task in reporting_tasks:
+        task_dict = task.to_dict()
+        task_component = task_dict["component"]
+        task_name = task_component["name"]
+        task_port = task_component["properties"]["prometheus-reporting-task-metrics-endpoint-port"]
+        if task_name == name and task_port == port:
+            return task
+
+    return None
+
+
+def create_reporting_task(name: str, port: str, version: str):
+    """Create a ReportingTask"""
+    task = nipyapi.nifi.models.reporting_task_entity.ReportingTaskEntity(
+        revision=nipyapi.nifi.models.revision_dto.RevisionDTO(version=0),
+        disconnected_node_acknowledged=False,
+        component=nipyapi.nifi.models.reporting_task_dto.ReportingTaskDTO(
+            name=name,
+            type="org.apache.nifi.reporting.prometheus.PrometheusReportingTask",
+            bundle=nipyapi.nifi.models.bundle_dto.BundleDTO(
+                group="org.apache.nifi",
+                artifact="nifi-prometheus-nar",
+                version=version
+            ),
+            properties={
+                "prometheus-reporting-task-metrics-endpoint-port": port,
+                "prometheus-reporting-task-metrics-send-jvm": True
+            }
+        )
+    )
+
+    controller_api = nipyapi.nifi.apis.controller_api.ControllerApi()
+
+    try:
+        return controller_api.create_reporting_task(body=task)
+    except Exception as ex:
+        raise Exception("Could not create reporting task: {}".format(str(ex))) from None
+
+
+def get_reporting_task_id(task):
```

Comment:
This function seems very brief, do we really need this?

Reply:
Yeah ill remove and just use task.id.